File: react-type-vite/src/services/api/request/sectionRequest.ts
import type { AssignmentRequest } from "./assignmentRequest";
import type { LessonRequest } from "./lessonRequest";
import type { QuizRequest } from "./quizRequest";

export interface SectionRequest {
  id?: number;
  courseId?: number; // Removed from backend but keeping for frontend logic
  title: string;
  description?: string;
  orderIndex?: number;
  isPublished?: boolean;
  lessons?: LessonRequest[];
  quizzes?: QuizRequest[];
  assignments?: AssignmentRequest[]
}

File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/request/SectionRequest.java
package com.hoangphihiep.dto.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;
import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SectionRequest {

    private Integer id;

    @NotBlank(message = "Section title is required")
    private String title; // Đổi từ sectionName thành title

    private String description;

    private Integer orderIndex;

    private Boolean isPublished;

    private List<Integer> visibleClassIds;

    private Set<LessonRequest> lessons;

    private Set<QuizRequest> quizzes;

    private Set<AssignmentRequest> assignments;

    private Date createdAt;

    private Date updateAt;
}
